scripts/set_gem_timing

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+NC_STRING="NC"
+SC_STRING="SC"
+# TODO:(josh) reflect on absolute paths
+IOC_PATH_NC="/reg/g/pcds/epics/ioc/fee/GasDetDAQ/R4.0.22"
+IOC_PATH_SC="/reg/g/pcds/epics/ioc/fee/GasDetDAQ/R4.0.22-NOTS"
+IOC_PATH=0
+
+# User input validation
+if [[ -n $1 ]]; then # if the user passed at least one arg
+  if [[ $1 != $NC_STRING ]] && [[ $1 != $SC_STRING ]]; then #validate argument is either SC or NC
+    echo "Error. Argument must be either: [NC or SC]"
+    exit 1
+  else
+    echo "User passed argument of: $1"
+  fi
+else
+  echo "Error. Must pass at least one argument"
+  exit 1
+fi
+
+if [[ -n $2 ]]; then
+  echo "Warning. Must pass strictly one argument, additional arg: $2 is ignored"
+  exit 1
+fi
+
+# check current operational state
+OP_MODE=$(caget KFE:CAM:TPR:02:MODE | awk '{print $2}')
+SET_MODE_CMD="caput KFE:CAM:TPR:02:MODE $1"
+
+echo "Current OP_MODE is: $OP_MODE"
+if [[ $OP_MODE = $1 ]]; then
+  echo "Already in requested mode: $OP_MODE exitting with NOP"
```

Comment:
Is it possible that the user might have gotten into an inbetween state where the PVs have been set to the requested mode, but the IOC has not been switched over yet? If so, I think we should proceed regardless instead of exiting here.

Reply:
I think this was my only important suggestion, and it's also the only one skipped over. I still maintain that we shouldn't immediately exit if the PV is already changed.

Reply:
I thought I had looked at that and somehow did not see it still did that. I agree that we should not quit when one of the PVs is changed because the IOC may not be changed yet. The cleanest way would be to get the running version and only update if it's not what we want in the requested mode.

Reply:
Thanks for following up on this, sorry for missing it. I'll make the patch Silke recommended.

Reply:
Just not exiting is the quicker solution, but it may cause additional IOC reboots which do take a while. But - safety by obscurity?

Reply:
Unrelated comment: currently you exit with the second argument, while the text just tells users it'll be ignored. Super minor.

Reply:
I see what you're saying. When I wrote this I treated the PV as source of truth and you are saying it should instead by IOC software because that has a longer boot time. From my work with the operators the past week they seem very competent, I'd imagine like you allude to above gating may actually induce additional edge cases. 

They seem happy to use this tool as more of a hammer or hack to clobber the HXR GEM into the timing schema they want. 

Exiting with the second argument was done to be explicit about what is ignored, when invoking CLI arguments I have found people sometimes have difficulty counting (may be a relic of `char * argv[]` things as a joke to myself).